spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/HyperConformance.scala

```diff
@@ -155,11 +159,18 @@ object HyperConformance extends StreamTransformerFactory with HyperConformanceAt
     }
   }
 
-  private def getReportDate(conf: Configuration): String = {
+  private def getReportDate(conf: Configuration): Column = {
     if (conf.containsKey(reportDateKey)) {
-      conf.getString(reportDateKey)
+      val reportDate = conf.getString(reportDateKey)
+      log.info(s"Information date: Explicit from the job configuration = $reportDate")
+      to_date(lit(reportDate), ReportDateFormat)
+    } else if (conf.containsKey(eventTimestampColumnKey)) {
+      val eventTimestampColumnName = conf.getString(eventTimestampColumnKey)
+      log.info(s"Information date: Derived from the event time column = $eventTimestampColumnName")
+      col(eventTimestampColumnName).cast(DateType)
```

Comment:
Can we be sure this would pass? Is it going to fail fast with a proper message?

Reply:
If the column does not exist or cannot be converted to date, it will fail on a Spark error.

Reply:
You probably want me to add validations of the existence of the column and its type?

Yeah, I should do it. Also, we will need a pattern if it is a string type.

Reply:
Yes. The thing is normal spark error would be lost (i think) and would look like normal std error. At least catch it and throw error with better message.